compiler/rustc_const_eval/src/interpret/cast.rs

```diff
@@ -305,6 +305,59 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         }
     }
 
+    fn unsize_just_metadata(
```

Comment:
This is mostly just copy/paste the above `fn usize_into_ptr` which doesn't feel great but deduplicating seems difficult.

Reply:
If we can factor that function into separately handling the data and meta part of the ptr, shouldn't that do the trick?

Reply:
The problem with doing it simply is that this handles both `Scalar` and `ScalarPair` pointers, so we can't "just" split the pointer. Or at the very least I don't know how to.

Reply:
I stand corrected, as complaining was enough to get me to find how to do it 🙃